### PR TITLE
Split project into modules

### PR DESCRIPTION
**Pull request by svilenvul**
Project split into individual modules for:
- the Maven plugin
- Checkstyle custom checks
- PMD Custom Checks
- FindBugs Custom checks

After the change the custom checks will be deployed separately
from the Maven plugin.

Additionally:
- Reduced scope of some dependencies to "provided";
- Removed some incorrect imports;
- Removed some transitive dependencies (commons-io, commons-lang,..).

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

**Comment by svilenvul** [comment]
Important notes :
-  this should not affect the builds using the Maven plugin, as the artifact ID is not changed. The Maven Plugin automatically adds the custom checks as dependencies;
- tested with the Eclipse SmartHome build;
- partially addresses https://github.com/openhab/static-code-analysis/issues/203, by isolating the dependencies in different modules.

**Comment by svilenvul** [comment]
@martinvw, it would be great, if you can take a look at this one :)

**Comment by martinvw** [comment]
Can you assign it to, I use that list to keep track of the most important things to look at.

**Comment by svilenvul** [comment]
Sorry, I do not see you in the list :-(

**Comment by svilenvul** [comment]
@martinvw, just a kind reminder :)

File: custom-checks/pom.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <name>Static Code Analysis Custom Checks</name>
+  
+  <modules>
+    <module>checkstyle</module>
+    <module>pmd</module>
+    <module>findbugs</module>
+  </modules>
+  
+  <packaging>pom</packaging>
+</project>
```
Comment (martinvw) [inline custom-checks/pom.xml]: Missing newline

**Comment by svilenvul** [comment]
Thanks !

**Pull request merged**